medical_prescription_sale_stock_us/models/medical_prescription_order_line.py

```diff
@@ -31,27 +30,31 @@ class MedicalPrescriptionOrderLine(models.Model):
         compute='_compute_can_dispense_and_qty',
         help='Total units allowed in the prescription, including refills',
     )
-    last_dispense_remain_qty = fields.Float(
+    dispense_remain_qty = fields.Float(
         string='Dispense Remaining Qty',
         compute='_compute_dispense_remain',
-        help='Estimated number of units remaining from last dispense',
-    )
-    last_dispense_remain_percent = fields.Float(
-        string='Dispense Remaining Percent',
-        compute='_compute_dispense_remain',
-        help='Estimated percentage remaining from last dispense',
+        help='Estimated number of units remaining from all finished dispenses',
     )
-    last_dispense_remain_day = fields.Float(
+    dispense_remain_day = fields.Float(
         string='Dispense Remaining Days',
         compute='_compute_dispense_remain',
-        help='Estimated days remaining from last dispense',
+        help='Estimated days remaining based on all finished dispenses',
     )
 
     @api.multi
+    @api.depends(
+        'duration_uom_id',
+        'duration',
+        'dispensed_ids',
+        'dispense_uom_id',
+        'qty',
+        'refill_qty_original',
+    )
     def _compute_dispense_remain(self):
-
         day_uom_id = self.env.ref('medical_medication.product_uom_day')
         for rec_id in self:
+            if len(rec_id.duration_uom_id) == 0 or rec_id.duration <= 0:
```

Comment:
Yeah. It definitely works. I already have a test case in place for this guard, and it still passes. Plus passing an empty recordset to `bool()` in console results in `False`.